cookbook/product/update-product.rst

```diff
@@ -4,96 +4,236 @@ How to Update Products
+
+    // sets a data in the product name (a field)
+    $propertySetter->setData($product, 'categories', ['category_code1', 'category_code2']);
+
+    // sets a localizable and scopable product attribute data
+    $propertySetter->setData($product, 'description', 'my description', ['locale' => 'en_US', 'scope' => 'mobile']);
+
+.. note::
+
+    This service has been reworked since the 1.3 to add the support of doctrine fields, become more complete and consistent with other property updaters.
+
+    The method 'ProductUpdater::setValue()' is now deprecated, you should use the `PropertySetterInterface::setData()`.
+
+Instanciate a New PropertyAdderInterface
+-----------------------------------------
+
+The product property adder implements the ``Akeneo\Component\StorageUtils\Updater\PropertyAdderInterface``.
+
+It's available has a service, you can fetch it from the container.
+
+.. code-block:: php
+
+    $propertyAdder = $this->getContainer()->get('pim_catalog.updater.product_property_adder');
 
-Setters
--------
+Use the PropertyAdderInterface
+------------------------------
 
-In order to use setters you will need to use the 'setValue()' method. The first argument matches all the products
-you want to set a new value, the second argument is the attribute code, the third argument is the value to set (it
-has to be compliant with the attribute type). Then you can specify the locale and the scope (locale and scope must be
-used when the attribute is localizable or scopable).
+The property adder allows to add an item in a single property (doctrine field data or attribute data) which contain a collection.
+
+This one is only available for properties that contains many items (as categories, options, prices).
+
+The property adder keep the existing items and add a new item inside.
 
 .. code-block:: php
 
-    $updater
-        ->setValue($products, 'name', 'Akeneo T-Shirt (new)')
-        ->setValue($products, 'description', 'Akeneo T-Shirt white with short sleeve (new)', 'en_US', 'ecommerce')
-        ->setValue($products, 'price', [['data' => 101.2, 'currency' => 'USD']], 'en_US', 'mobile');
+    // adds a data in the product color (a multiselect attribute)
+    $propertyAdder->addData($product, 'color', ['red']);
+
+    // adds a data in the product categories (a field)
+    $propertyAdder->addData($product, 'category', ['tshirt', 'promo']);
+
+Instanciate a New PropertyRemoverInterface
+------------------------------------------
+
+The product property remover implements the ``Akeneo\Component\StorageUtils\Updater\PropertyRemoverInterface``.
+
+It's available has a service, you can fetch it from the container.
+
+.. code-block:: php
+
+    $propertyRemover = $this->getContainer()->get('pim_catalog.updater.product_property_remover');
+
+Use the PropertyRemoverInterface
+--------------------------------
+
+The property remover allows to remove an item from a single property (doctrine field data or attribute data) which contain a collection.
```

Comment:
i hope they do the diff between doctrine fields or attribute value but this should be detailed in another chapter